examples/www-image/main/jpg-render.c

```diff
@@ -0,0 +1,518 @@
+#include "esp_event.h"
+#include "esp_log.h"
+#include "nvs_flash.h"
+#include "lwip/err.h"
+#include "lwip/sys.h"
+// HTTP Client
+#include "esp_netif.h"
+#include "esp_tls.h"
+#include "esp_http_client.h"
+#include "esp_crt_bundle.h"
+
+// JPG decoder
+#if ESP_IDF_VERSION_MAJOR >= 4 // IDF 4+
+  #include "esp32/rom/tjpgd.h"
+#else // ESP32 Before IDF 4.0
+  #include "rom/tjpgd.h"
+#endif
+
+#include "esp_task_wdt.h"
+#include <stdio.h>
+#include <string.h>
+#include <math.h> // round + pow
+
+#include "epd_driver.h"
+#include "epd_highlevel.h"
+EpdiyHighlevelState hl;
+
+// WiFi configuration
+#define ESP_WIFI_SSID     "WLAN-724300"
+#define ESP_WIFI_PASSWORD "50238634630558382093"
+
+
+// Image URL and jpg settings. Make sure to update WIDTH/HEIGHT if using loremflickr
+// Note: Only HTTP protocol supported (Check README to use SSL secure URLs)
+#define IMG_URL "https://loremflickr.com/960/540"
+//#define IMG_URL "http://img.cale.es/jpg/fasani/5e636b0f39aac"
+
+// Jpeg: Adds dithering to image rendering (Makes grayscale smoother on transitions)
+#define JPG_DITHERING true
+// Affects the gamma to calculate gray (lower is darker/higher contrast)
+// Nice test values: 0.9 1.2 1.4 higher and is too bright
+double gamma_value = 0.9;
+
+// As default is 512 without setting buffer_size property in esp_http_client_config_t
+#define HTTP_RECEIVE_BUFFER_SIZE 1536
+extern const uint8_t server_cert_pem_start[] asm("_binary_server_cert_pem_start");
+extern const uint8_t server_cert_pem_end[] asm("_binary_server_cert_pem_end");
+// EPD Waveform
+#define WAVEFORM EPD_BUILTIN_WAVEFORM
+// Minutes that goes to deepsleep after rendering
+// If you build a gallery URL that returns a new image on each request (like cale.es)
+// this parameter can be interesting to make an automatic photo-slider
+#define DEEPSLEEP_MINUTES_AFTER_RENDER 59
+
+#define DEBUG_VERBOSE true
+
+// JPEG decoder
+JDEC jd; 
+JRESULT rc;
+// Buffers
+uint8_t *fb;            // EPD 2bpp buffer
+uint8_t *source_buf;    // JPG download buffer
+uint8_t *decoded_image; // RAW decoded image
+static uint8_t tjpgd_work[4096]; // tjpgd 4Kb buffer
+
+uint32_t buffer_pos = 0;
+uint32_t time_download = 0;
+uint32_t time_decomp = 0;
+uint32_t time_render = 0;
+static const char * jd_errors[] = {
+    "Succeeded",
+    "Interrupted by output function",
+    "Device error or wrong termination of input stream",
+    "Insufficient memory pool for the image",
+    "Insufficient stream input buffer",
+    "Parameter error",
+    "Data format error",
+    "Right format but not supported",
+    "Not supported JPEG standard"
+};
+
+const uint16_t ep_width = EPD_WIDTH;
+const uint16_t ep_height = EPD_HEIGHT;
+uint8_t gamme_curve[256];
+
+static const char *TAG = "EPDiy";
+uint16_t countDataEventCalls = 0;
+uint32_t countDataBytes = 0;
+uint32_t img_buf_pos = 0;
+uint32_t dataLenTotal = 0;
+uint64_t startTime = 0;
+
+/*====================================================================================
```

Comment:
Extend this notice?

Reply:
I don’t know if I really got what had to be done here but I extended the comment leaving the credits to abdomen